5/7/19 functional+cosmetic changes

Accio1 opened the pull request:
@locness3 This is my most recent update. The change log is below. I figured you might want to review it. If you don't, that's fine as well.

Change Log:
javascript.js (Renamed later to main.js):
-Added a 404 page for if the user inputs an invalid Studio Id
-Added two shortcuts (SDS and Featured), which direct to the current SDS and Featured Studio

stylesheet.css:
-Added CSS for the error box
-Added min-height in order to make sure the box doesn't get too small

index.html:
-Added the place where an error message can be displayed.
-Added a default title of "Loading..."

(New File) home/home.js:
Moved JavaScript from script tags to a separate file

home/stylesheet.css:
-Added style for DevTag
-Removed blue glow from embed code textarea
-Changed <input type="submit"> to buttons to make things more consistent.

home/index.html:
-Added a version tag (Currently says STABLE)
-Made the current featured studio the default studio displayed
-Made text appear whenever someone copies something
-Move JavaScript to a separate file (home.js)
-Removed most CSS from <head> element.

Accio1 committed:
Add 404 + Shortcuts

-Added a 404 page for if the user inputs an invalid Studio Id
-Added two shortcuts (SDS and Featured), which direct to the current SDS and Featured Studio

Accio1 committed:
Small Cosmetic Changes

-Added CSS for the error box
-Added min-height in order to make sure the box doesn't get too small

Accio1 committed:
Added error + default title

-Added the <p> element for when there is an error
-Added a default title of "Loading..."

Accio1 committed:
Create home.js

Moved JavaScript from script tags to a separate file

Accio1 committed:
Small Cosmetic Changes

-Added style for DevTag
-Removed blue glow from embed code textarea
-Changed <input type="submit"> to buttons to make things more consistent.

Accio1 committed:
Default featured studio + Copied Box + Separate JS

-Added a version tag (Currently says STABLE)
-Made the current featured studio the default studio displayed
-Made text appear whenever someone copies something
-Move JavaScript to a separate file (home.js)
-Removed most CSS from <head> element.

Accio1 committed:
Rename javascript.js to main.js

moeenio reviewed (changes requested):
The script doesn't load on home page, because it's pointing to the old javascript.js

Accio1 committed:
Update index.html

Accio1 commented:
> The script doesn't load on home page, because it's pointing to the old javascript.js

Fixed

Accio1 commented:
Is everything else fine?

moeenio commented:
I'm testing...

moeenio commented:
The way you list changelog by file in your PR description is weird. Do a changelog per feature.

moeenio commented on home/home.js:
```diff
@@ -0,0 +1,41 @@
+			document.getElementById("GT_result").innerHTML = "Not A Studio ID";
+  			} 
+		else {
+    			window.location.href = "/SSE/#"+GT_StudioID;
```
Use `..` instead of `/SSE` (for testing locally or with Netlify)

Accio1 replied:
Done

moeenio commented on home/index.html:
```diff
@@ -8,83 +8,35 @@
 		<div class="content">
-			<h1>Scratch Studio Embed</h1>
+			<h1>Scratch Studio Embed
+			<span id="DevTag">STABLE</span></h1>
```
Logically, the easier accessible version is the stable one. You should either remove that, or start using the GitHub 's releases feature and number versionning (following the model MAJOR.MINOR.PATCH), and indicate the release number here.

Accio1 replied:
Fixed, I made it display "By -Accio-" instead of "STABLE"

Accio1 committed:
Fix relative link

Accio1 committed:
Change DevTag

Change DevTag from"STABLE" to "By -Accio-"

Accio1 commented:
Is it now ready to be merged?

Accio1 committed:
Fixed Short Link

moeenio commented on main.js:
```diff
@@ -0,0 +1,49 @@
+setTimeout(display, 50)
+
+if (window.location.hash === "" || window.location.hash === "#") {
+	window.location.replace("/SSE/home");
```
Same here, don't add `/SSE`. in this case you can use `home` directly.

Accio1 replied:
That doesn't work. It brings me to https://accio1.github.io/home

moeenio replied:
Without a `/` at the beginning. or try `./home`

Accio1 replied:
I added a base element and made it just "home". I'm assuming this is the last thing that needed to be changed, as my comment below said:

"If it's not ready, please suggest everything which needs to be changed all at once next time, as then I know I can merge as soon as I have implemented the suggestions."

If there are no other suggestions within two hours, I will assume that was the last one and merge this.

Thanks for all of your help!

moeenio replied:
Wait!

Accio1 commented:
Can it be merged?

Accio1 commented:
> Can it be merged?

If it's not ready, please suggest everything which needs to be changed all at once next time, as then I know I can merge as soon as I have implemented the suggestions.

Accio1 committed:
Add <base> tag

Accio1 committed:
Add <base> tag

Accio1 committed:
Add <base> tag

Accio1 committed:
Add <base> tag

moeenio reviewed (changes requested):
The idea with using relative paths is to **_never_** have an absolute URL. I just can't test locally, and you can't too with what we have (Did you even **try** testing it locally?). Everything is redirected to the current GitHub version. You should use relative paths **_everywhere_**. In relative paths, use :
- `/` for root
- `.` for current folder
- `..` for parent folder
- nothing for the current folder

Accio1 commented:
I did test it locally, by combining the stylesheet and the JavaScript into the index.html. Then I separated it when I put it on GitHub.

moeenio commented:
Well, you shouldn't do that! You should use relative urls, so everyone can test it anywhere without playing around! I gave you a full guide in my review.

Accio1 commented:
That doesn’t work on Chromebooks. That’s why I do it the way I do.

moeenio commented:
Deploy to Netlify then. All you have to do is to drag your site folder.

Accio1 commented:
I’d rather just test it locally on my Chromebook.

moeenio commented:
Please, test it on Netlify. To make sure your relative urls and all work correctly, and to make sure it will work perfectly on GitHub pages.

moeenio commented:
You can register on Netlify with your GitHub account, and once you have done so, you just have to **drag-and-drop the folder on Netlify's interface** to deploy your site.

Accio1 commented:
Yes, but I don’t mind just testing it on my Chromebook.

moeenio commented:
That's bad practice. Sorry, but you really need to test it on a "real" environement. You can't ask people to review your changes if they can't do it the way they would do it. And you shouldn't rely on your GitHub URL for flexibility.

Accio1 committed:
Change to relative links

Accio1 committed:
Update index.html

Accio1 committed:
Update home.js

Accio1 committed:
Update home.js

Accio1 committed:
Update home.js

Accio1 commented:
You've convinced me. I made all of the URLs relative. Are you able to test it now?

moeenio commented:
Yes! It works great

moeenio commented:
As I already said, the way you do your changelog by file in your PR description is weird. Do a changelog per feature.

Accio1 commented:
Change Log: 
-Added a 404 page if the studio id is invalid.
-Added shortcuts which can replace studio ids (Featured for the current featured studio, SDS for the current SDS).
-Added a min-height style property which makes sure the box is always a certain size.
-Made the default title "Loading..." so users know that it's loading.
-Moved the JavaScript for the homepage into a separate file instead of script tags.
-Changed `<input type="submit">` to buttons to make things more consistent.
-Added a little bubble next to the title which says "By: -Accio-"
-Added some text which appears when someone uses the copy button

Ready to merge now?

Accio1 commented:
I saw that you approved the changes! Merging!

moeenio commented:
🎉

moeenio commented:
BTW you could have edited your PR's description isntead of creating a new comment, but that's okay
